### DISCUSSION TITLE
How can I automate creation of a default account (or configure that in a config file)?

### DISCUSSION THREAD

**Discussion by captnswing** [lightdash/lightdash#1359] [Q&A] [answered]
Using lightdash (and postgresql) in docker, I don't want to manually re-enter account details and dbt configuration through Web UI - can I just put those details in a config file?

The only way I see to do this now is to go through manual account creation and dbt configuration steps once, and then dump the relevant rows from the psql tables (which ones?) to insert them automatically the next time the psql database is reset

![CleanShot 2022-02-12 at 09 02 16](https://user-images.githubusercontent.com/325377/153702957-f34a684f-f90d-4017-923e-1e4fc29e78b9.png)

**Comment by ZeRego** [comment, marked as answer]
Good question @captnswing ! 

We do have plans to [support everything as code ](https://github.com/lightdash/lightdash/issues/369) but we aren't there yet. Would be great if you could explained your use case under that thread so we can take it into account when implementing the feature :) 

Right now, you could use the existing REST API even though we haven't made it public ( [we have a ticket for it](https://github.com/lightdash/lightdash/issues/632) ). Using the REST API has a higher chance of future compatibility compared with manipulating the DB directly.  

If using the API, it would involve making 3 HTTP requests:
1. Create user
```
[POST] /api/register
body: {
    firstName: string;
    lastName: string;
    email: string;
    password: string
}
```
3. Complete user creation
```
[PATCH] /api/user/me/complete
headers: cookie ( using cookie returned from first request )
body: {
    organizationName: string;
    jobTitle: string;
    isMarketingOptedIn: boolean;
    isTrackingAnonymized: boolean;
}
```
3. Connect project
```
[POST] /api/org/projects
headers: cookie ( using cookie returned from first request )
body: {
    name: string;
    dbtConnection: DbtProjectConfig;
    warehouseConnection?: WarehouseCredentials;
}
```
The warehouse credentials is completely different depending on the warehouse you want to use. To check the correct structure I recommend checking [packages/common/src/index.ts](https://github.com/lightdash/lightdash/blob/main/packages/common/src/index.ts#L1042) or let us know what warehouse you use.
The same for the dbt project configuration, let us know if you can't find the correct data format in common/src/index.ts.
You can also copy the requests when using the browser.

**Reply by captnswing** [reply]
Thank you!
